pkg/operator/ceph/cluster/controller.go

```diff
@@ -146,41 +161,57 @@ func add(opManagerContext context.Context, mgr manager.Manager, r reconcile.Reco
 	}
 
 	// Watch for changes on the CephCluster CR object
-	s := source.Kind[client.Object](
-		mgr.GetCache(), &cephv1.CephCluster{TypeMeta: ControllerTypeMeta},
-		&handler.EnqueueRequestForObject{}, watchControllerPredicate(opManagerContext, mgr.GetClient()),
+	err = c.Watch(
+		source.Kind(
+			mgr.GetCache(),
+			&cephv1.CephCluster{TypeMeta: ControllerTypeMeta},
+			&handler.TypedEnqueueRequestForObject[*cephv1.CephCluster]{},
+			watchControllerPredicate(opManagerContext, mgr.GetClient()),
+		),
 	)
-	err = c.Watch(s)
 	if err != nil {
 		return err
 	}
 
 	// Watch all other resources of the Ceph Cluster
 	for _, t := range objectsToWatch {
-		err = c.Watch(
-			source.Kind[client.Object](
-				mgr.GetCache(), t, handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &cephv1.CephCluster{}),
-				opcontroller.WatchPredicateForNonCRDObject(&cephv1.CephCluster{TypeMeta: ControllerTypeMeta}, mgr.GetScheme())),
-		)
+		err = watchOwnedCoreObject(c, mgr, t)
```

Comment:
This function is defined but then only used here. It seems like it would be more straightforward to inline this here the same as the other watches, unless I'm missing something with the looping